src/libraries/System.Private.CoreLib/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -38,10 +36,6 @@ internal enum BufferDroppedReason : int
             OverMaximumSize,
         }
 
-        // Parameterized constructor to block initialization and ensure the EventSourceGenerator is creating the default constructor
-        // as you can't make a constructor partial.
-        private ArrayPoolEventSource(int _) { }
```

Comment:
@benaadams's whole purpose for this ctor (which should never actually be used and should be trimmed away) was to cause there to a build error if generator somehow didn't run.  Now that you've moved this into the generator, doesn't that defeat the purpose?

Reply:
Was a workaround because while you can mark methods as `partial` to get same effect you can't mark constructors as `partial`, so there's no built in way to error if the generator doesn't run; hence this parameterized constructor stub which will fail if the generator doesn't create the default no parameter constructor.

Reply:
Right.  So if this is moved into the generator, the benefits of that workaround go away. Unless I'm missing something then, either this PR should be closed or we should just delete these dummy ctors, but generating them seems pointless.

Reply:
Yeah and if the default constructor is not generated; then its going to use a nop "default" constructor that won't setup the eventsource (which is why I wanted to force it being created or fail)

Reply:
> Unless I'm missing something then, either this PR should be closed or we should just delete these dummy ctors, but generating them seems pointless.

The constructor is used, see the newly generated code example https://github.com/dotnet/runtime/pull/51978/files#diff-5ebf48679be706fe4d112ad18084009ccb8620e060b8c146899d0f0ec7bc940dR33

Reply:
> The constructor is used, see the newly generated code example

To what benefit?

Reply:
To avoid constructing EventSource object with useless data when EventSource is not supported

Reply:
But what does it actually save? If it's unsupported,  the code in the ctor that's otherwise delegated to is trimmed away.

Reply:
The generated constructor payload (e.g. https://github.com/dotnet/runtime/pull/51978/files#diff-5ebf48679be706fe4d112ad18084009ccb8620e060b8c146899d0f0ec7bc940dR29) and eventually System.Guid

Reply:
The arguments? Those are minimal. I don't think this is worthwhile.